core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -702,22 +712,26 @@ class KafkaController(val config: KafkaConfig,
               topicDeletionManager.markTopicIneligibleForDeletion(Set(topic),
                 reason = "topic reassignment in progress")
               onPartitionReassignment(tp, reassignedPartitionContext)
+              reassignmentResults.put(tp, ApiError.NONE)
             } catch {
               case e: ControllerMovedException =>
                 error(s"Error completing reassignment of partition $tp because controller has moved to another broker", e)
                 throw e
               case e: Throwable =>
                 error(s"Error completing reassignment of partition $tp", e)
                 partitionsToBeRemovedFromReassignment.add(tp)
+                reassignmentResults.put(tp, new ApiError(Errors.UNKNOWN_SERVER_ERROR))
             }
           }
         } else {
             error(s"Ignoring request to reassign partition $tp that doesn't exist.")
             partitionsToBeRemovedFromReassignment.add(tp)
+            reassignmentResults.put(tp, new ApiError(Errors.UNKNOWN_TOPIC_OR_PARTITION, "The partition does not exist."))
         }
       }
     }
     removePartitionsFromReassignedPartitions(partitionsToBeRemovedFromReassignment)
```

Comment:
Do we want to guarantee that if `controllerContext.partitionReplicaAssignment(tp).isBeingReassigned` then `controllerContext.partitionsBeingReassigned(tp).isPresent`?
At this point, I don't know if we should update the in-memory and ZK replica assignment to one that is not being reassigned. Since we remove the partition from the `partitionsBeingReassigned` collection, it seems to make sense to also update in-memory and ZK to truly treat the reassignment as failed.
Yet, do we need to send LeaderAndIsr requests if we change those?

The one scenario where this divergence could happen (`controllerContext.partitionReplicaAssignment(tp).isBeingReassigned` is true and `controllerContext.partitionsBeingReassigned(tp).isPresent` is not)
is if we have an UNKNOWN_SERVER_ERROR during the reassignment (line 723)